stackstorm-compose/docker-compose.yml

```diff
@@ -0,0 +1,194 @@
+version: '3'
+
+services:
+  st2web:
```

Comment:
Per #k8s discussion, to fix the nginx DNS issues described in #29 and #30

We'll need adding healthchecks for `st2web`: https://docs.docker.com/compose/compose-file/#healthcheck
And so `st2web` container will get forcefully restarted once it begins showing any DNS issues due to changed hostnames.
This would also conform with the K8s story as we have a liveness probe check for st2web.
See https://github.com/StackStorm/stackstorm-ha/blob/master/templates/deployments.yaml#L364-L370 in K8s.
That would fix an issue for docker-compose too.

Reply:
I also did some research and totally agree with @armab. 

So the actual cause is that the st2web container starts way faster during a restart and is up again before st2stream and st2auth for example went down. So they get a new IP during the startup which is too late because nginx is already started and did the DNS lookup.

So let's work on the healthchecks. First tests in my lab were successful.

Reply:
I'm looking into healthchecks as well and I agree. I think st2api, st2stream, and st2auth should have a healtheck themselves, which should be easy. The tricky part is again, st2web because we need to check 3 urls.  st2api will return a 401 Unauthorized.

Reply:
I've been looking to docker health checks. In fact, I've created an st2web-healthchec.sh file (code below comment) that is mounted and called.  It will mark the container as unhealthy.

However - this will not help. [Docker containers do not exit on unhealthy status](https://stackoverflow.com/questions/47088261/restarting-an-unhealthy-docker-container-based-on-healthcheck).  

The best option I can think of (without converting over to docker swarm) is to change my `exit 1` to `nginx -s stop`.  That would kill the nginx process, stopping the container, and then docker would restart it.

Thoughts?


```bash
#!/usr/bin/env bash

API_STATUS=$(curl --write-out "%{http_code}\n" --silent --output /dev/null http://localhost/api/v1/)
if [ "${API_STATUS}" != "401" ]; then  echo "st2api failure";  exit 1; fi

STREAM_STATUS=$(curl --write-out "%{http_code}\n" --silent --output /dev/null http://localhost/stream/v1/stream)
if [ "${STREAM_STATUS}" != "401" ]; then  echo "st2stream failure";  exit 1; fi

AUTH_STATUS=$(curl --write-out "%{http_code}\n" --silent --output /dev/null  http://localhost/auth/v1/)
if [ "${AUTH_STATUS}" != "404" ]; then  echo "st2auth failure";  exit 1; fi

exit 0
```

Updated entry in docker-compose:

```docker-compose
  st2web:
    image: ${ST2_IMAGE_REPO:-stackstorm/}st2web:${ST2_VERSION:-3.3dev}
    restart: always
    environment:
      ST2_AUTH_URL: ${ST2_AUTH_URL:-http://st2auth:9100/}
      ST2_API_URL: ${ST2_API_URL:-http://st2api:9101/}
      ST2_STREAM_URL: ${ST2_STREAM_URL:-http://st2stream:9102/}
      ST2WEB_HTTPS: ${ST2WEB_HTTPS:-0}
    depends_on:
      - st2auth
      - st2api
      - st2stream
    healthcheck:
      test: ["CMD", "/st2web-healthcheck.sh"]
      interval: 30s
      timeout: 1s
      retries: 1
    volumes:
      - ./st2web-healthcheck.sh:/st2web-healthcheck.sh
    ports:
      - "${ST2_EXPOSE_HTTP:-127.0.0.1:8000}:80"
      # - "${ST2_EXPOSE_HTTPS:-127.0.0.1:8443}:443"
      # more work would be needed with certificate generate to make https work.
    networks:
      - private
      - public
    dns_search: .
```

Reply:
I think it's an interesting idea with the nginx stop. Let's try it.

Reply:
Changes pushed.

Reply:
@winem Can you confirm the fix worked for the issue you reported?

Reply:
Sorry, I was busy the last few days but now I can confirm that it works as expected and the issue is no longer reproducible. Great job!